ArjunaJTA/jta/classes/com/arjuna/ats/jta/common/JTAEnvironmentBean.java

```diff
@@ -595,19 +596,22 @@ public void setXaResourceOrphanFilters(List<XAResourceOrphanFilter> xaResourceOr
      *
      * @return true for cleanup during prepare, false for cleanup during phase two rollback.
      */
+    @Deprecated
     public boolean isXaRollbackOptimization()
     {
-        return xaRollbackOptimization;
+        return false;
     }
 
     /**
+     * Deprecated: This method will be removed in the next major release
+     *
      * Sets if failed resources should be cleaned up during prepare or during phase two.
      *
      * @param xaRollbackOptimization true for immediate cleanup, false for phase two cleanup.
      */
+    @Deprecated
     public void setXaRollbackOptimization(boolean xaRollbackOptimization)
     {
-        this.xaRollbackOptimization = xaRollbackOptimization;
```

Comment:
If removing the functionality does go ahead, I think logging a warning might (needs a bit more thought) be appropriate to add here

Reply:
Thanks Tom. Done

Reply:
Thank you for adding the warning